lib/Feature/FeatureModelWriter.cpp

```diff
@@ -184,12 +185,25 @@ int FeatureModelXmlWriter::writeBooleanConstraints( // NOLINT
     CHECK_RC
   }
 
-  // TODO write other boolean constraints if they are parsed
+  // TODO(se-passau/VaRA#664): write other boolean constraints if they are
+  //  parsed
 
   RC = xmlTextWriterEndElement(Writer); // BOOLEANCONSTRAINT
   return RC;
 }
 
+OrderedFeatureVector getChildrenFeatures(FeatureTreeNode *N) {
+  OrderedFeatureVector V;
+  for (FeatureTreeNode *C : N->children()) {
+    if (auto *F = llvm::dyn_cast<Feature>(C); F) {
+      V.insert(F);
+    } else {
+      V.insert(getChildrenFeatures(C));
+    }
+  }
+  return std::move(V);
```

Comment:
```suggestion
  return V;
```
If you return a local variable in all return paths C++17 guarantees return-value optimization, which can directly constructs the value in the caller stack frame. Using move here would prevent RVO.